--- vlc-3.0.10/modules/gui/qt/dialogs/plugins.hpp~	2017-11-24 16:29:17.000000000 +0100
+++ vlc-3.0.10/modules/gui/qt/dialogs/plugins.hpp	2020-06-02 09:11:52.586760706 +0200
@@ -32,6 +32,7 @@
 
 #include <QStringList>
 #include <QTreeWidgetItem>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QAbstractListModel>
 #include <QSortFilterProxyModel>
--- vlc-3.0.10/modules/gui/qt/components/playlist/views.hpp~	2017-11-24 16:29:17.000000000 +0100
+++ vlc-3.0.10/modules/gui/qt/components/playlist/views.hpp	2020-06-02 09:13:07.692655049 +0200
@@ -26,6 +26,7 @@
 
 #include <QStyledItemDelegate>
 #include <QListView>
+#include <QPainterPath>
 #include <QTreeView>
 #include <QAbstractItemView>
 #include "util/pictureflow.hpp"
--- vlc-3.0.10/modules/gui/qt/util/timetooltip.hpp~	2018-05-03 14:03:21.000000000 +0200
+++ vlc-3.0.10/modules/gui/qt/util/timetooltip.hpp	2020-06-02 09:14:04.892066840 +0200
@@ -24,6 +24,7 @@
 
 #include "qt.hpp"
 
+#include <QPainterPath>
 #include <QWidget>
 
 class TimeTooltip : public QWidget
