--- vlc-0.8.1/configure.ac.orig	2005-03-30 03:06:33.000000000 +0200
+++ vlc-0.8.1/configure.ac	2005-03-30 03:07:15.000000000 +0200
@@ -192,7 +192,7 @@
 dnl
 dnl Gettext stuff
 dnl
-ALL_LINGUAS="da de en_GB es fr hu it ja nl no pl pt_BR ru sv"
+ALL_LINGUAS="da de en_GB es fr hu it ja nl nb pl pt_BR ru sv"
 AM_GNU_GETTEXT_VERSION(0.11.5)
 AM_GNU_GETTEXT
 if test "${nls_cv_force_use_gnu_gettext}" = "yes"; then
