--- vlc-0.8.1/configure.ac.orig	2005-03-30 03:06:33.000000000 +0200
+++ vlc-0.8.1/configure.ac	2005-03-30 03:07:15.000000000 +0200
@@ -192,7 +192,7 @@
 dnl
 dnl Gettext stuff
 dnl
-ALL_LINGUAS="bn da de el en_GB es fr hi hu it ja l33t my ne nl no pl ps pt_BR ru sv tet tl"
+ALL_LINGUAS="bn da de el en_GB es fr hi hu it ja l33t my ne nl nb pl ps pt_BR ru sv tet tl"
 AM_GNU_GETTEXT_VERSION(0.11.5)
 AM_GNU_GETTEXT
 if test "${nls_cv_force_use_gnu_gettext}" = "yes"; then
