--- vlc-0.8.6f/modules/access/cdda/cdda.h.orig	2008-04-04 23:27:12.870918273 +0200
+++ vlc-0.8.6f/modules/access/cdda/cdda.h	2008-04-04 23:27:47.868512969 +0200
@@ -75,7 +75,7 @@
   paranoia_none    = 0, /* Note: We make use of 0 as being the same as false */
   paranoia_overlap = 1, 
   paranoia_full    = 2
-} paranoia_mode_t;
+} vlc_paranoia_mode_t;
 
   
 /*****************************************************************************
@@ -107,7 +107,7 @@
 
 #if LIBCDIO_VERSION_NUM >= 72
   /* Paranoia support */
-  paranoia_mode_t e_paranoia;         /* Use cd paranoia for reads? */
+  vlc_paranoia_mode_t e_paranoia;         /* Use cd paranoia for reads? */
   cdrom_drive_t *paranoia_cd;         /* Place to store drive
 					 handle given by paranoia. */
   cdrom_paranoia_t *paranoia;
