--- vlc-3.0.11.1/modules/access/live555.cpp.orig	2019-03-29 20:01:15.000000000 +0100
+++ vlc-3.0.11.1/modules/access/live555.cpp	2021-01-19 06:23:21.598574547 +0100
@@ -852,7 +852,9 @@
             if( !p_sys->b_multicast )
             {
                 /* We need different rollover behaviour for multicast */
-                p_sys->b_multicast = IsMulticastAddress( sub->connectionEndpointAddress() );
+                struct sockaddr_storage sa;
+                sub->getConnectionEndpointAddress( sa );
+                p_sys->b_multicast = IsMulticastAddress( sa );
             }
 
             tk = (live_track_t*)malloc( sizeof( live_track_t ) );
