diff -ur vlc-0.9.3.orig/modules/codec/x264.c vlc-0.9.3/modules/codec/x264.c
--- vlc-0.9.3.orig/modules/codec/x264.c	2008-09-23 15:41:30.000000000 -0700
+++ vlc-0.9.3/modules/codec/x264.c	2008-10-07 22:27:22.863190593 -0700
@@ -1082,7 +1082,7 @@
         p_sys->param.analyse.i_trellis = val.i_int;
 #endif
 
-#if X264_BUILD >= 41
+#if X264_BUILD >= 41 && X264_BUILD < 65
     var_Get( p_enc, SOUT_CFG_PREFIX "b-rdo", &val );
     p_sys->param.analyse.b_bframe_rdo = val.b_bool;
 #endif
@@ -1092,7 +1092,7 @@
     p_sys->param.analyse.b_fast_pskip = val.b_bool;
 #endif
 
-#if X264_BUILD >= 43
+#if X264_BUILD >= 43 && X264_BUILD < 65
     var_Get( p_enc, SOUT_CFG_PREFIX "bime", &val );
     p_sys->param.analyse.b_bidir_me = val.b_bool;
 #endif
