diff -uNr vlc-0.3.0.orig/src/interface/main.c vlc-0.3.0/src/interface/main.c
--- vlc-0.3.0.orig/src/interface/main.c	Sat Apr  6 07:54:37 2002
+++ vlc-0.3.0/src/interface/main.c	Wed Sep 11 14:40:43 2002
@@ -1316,7 +1316,7 @@
 
     i_capabilities |= CPU_CAPABILITY_FPU;
 
-#   ifdef CAN_COMPILE_ALTIVEC
+#   ifdef CAN_COMPILE_C_ALTIVEC
     signal( SIGILL, IllegalSignalHandler );
 
     psz_capability = "AltiVec";
