--- vlc-1.1.10/modules/text_renderer/freetype.c~	2011-06-06 01:31:20.000000000 +0200
+++ vlc-1.1.10/modules/text_renderer/freetype.c	2011-07-05 08:35:55.978852297 +0200
@@ -67,9 +67,9 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-# define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/vlc/skins2/fonts/FreeSans.ttf"
+# define SYSTEM_DEFAULT_FAMILY "Sans"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/LiberationMono-Regular.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
 #endif
 
