diff -urN vlc-0.2.61.org/configure.in vlc-0.2.61/configure.in
--- vlc-0.2.61.org/configure.in	Sun Feb 18 12:34:33 2001
+++ vlc-0.2.61/configure.in	Sun Feb 18 13:03:48 2001
@@ -168,12 +168,12 @@
     then
       LIB_SDL="-l"$withval
     else
-      LIB_SDL="-lSDL"
+      LIB_SDL="-L/usr/X11R6/lib -lSDL"
     fi ])
     if test "x$withval" == "x";
     then
       PLUGINS=${PLUGINS}"sdl "
-      LIB_SDL="-lSDL"
+      LIB_SDL="-L/usr/X11R6/lib -lSDL"
     fi
 AC_ARG_WITH(glide,
   [  --with-glide[=name]     Glide (3dfx) support (default disabled)],
