
--- vlc-0.8.1337/configure.ac.orig	2005-04-07 22:24:53.881869064 +0200
+++ vlc-0.8.1337/configure.ac	2005-04-07 22:28:02.355216760 +0200
@@ -3600,18 +3600,15 @@
   WXWINDOWS_NAME="wx-config"
   AC_ARG_WITH(wx-config,
     [    --with-wx-config=NAME      wx-config name (default is wx-config)],
+  # look for wx-config
     [ if test "${with_wx_config}" != "no"
       then
-        WXWINDOWS_NAME="${with_wx_config}"
-      fi ])
-  # look for wx-config
+  AC_PATH_PROG(WX_CONFIG, ${with_wx_config}, no)
+		else
   AC_PATH_PROG(WX_CONFIG, ${WXWINDOWS_NAME}, no, ${WXWINDOWS_PATH})
+      fi ])
   if test "${WX_CONFIG}" != "no" -a "${CXX}" != ""
   then
-    if expr 2.3.0 \> `${WX_CONFIG} --version` >/dev/null
-    then
-      AC_MSG_ERROR([Your development package for wxWindows is too old, you need at least version 2.3.0. Please upgrade and try again. Alternatively you can also configure with --disable-wxwindows.])
-    fi
     AC_LANG_PUSH(C++)
     # Turn this error:
     #   playlist.cpp:1351: error: ISO C++ forbids cast to non-reference type
