--- vlc-2.0.7/modules/demux/live555.cpp.orig	2013-02-08 13:31:21.000000000 +0100
+++ vlc-2.0.7/modules/demux/live555.cpp	2013-07-27 07:11:57.437238145 +0200
@@ -232,7 +232,7 @@
                    char const* applicationName, portNumBits tunnelOverHTTPPortNum,
                    demux_sys_t *p_sys) :
                    RTSPClient( env, rtspURL, verbosityLevel, applicationName,
-                   tunnelOverHTTPPortNum )
+                   tunnelOverHTTPPortNum, -1 )
     {
         this->p_sys = p_sys;
     }
