Index: vlc-0.8.2/configure.ac
===================================================================
--- vlc-0.8.2.orig/configure.ac
+++ vlc-0.8.2/configure.ac
@@ -1877,16 +1877,9 @@ if test "${enable_mkv}" != "no" -a "${CX
               if test "${SYS}" = "darwin"; then
                 VLC_ADD_CXXFLAGS([mkv],[-O1])
               fi
-              AC_CHECK_LIB(ebml_pic, main, [
-                # We have ebml_pic, that's good, we can build an mkv.so plugin !
+              AC_CHECK_LIB(ebml, main, [
                 VLC_ADD_PLUGINS([mkv])
-                VLC_ADD_LDFLAGS([mkv],[-lmatroska_pic -lebml_pic])
-              ], [
-                AC_CHECK_LIB(ebml, main, [
-                  # We only have libebml, make mkv.a a builtin
-                  VLC_ADD_BUILTINS([mkv])
-                  VLC_ADD_LDFLAGS([mkv],[-lmatroska -lebml])
-                ])
+                VLC_ADD_LDFLAGS([mkv],[-lmatroska -lebml])
               ])
             ],
             [AC_MSG_RESULT([no])
