diff -x '*~' -ur vlc-snapshot-20050607/src/misc/cpu.c vlc-snapshot-20050607-amd64/src/misc/cpu.c
--- vlc-snapshot-20050607/src/misc/cpu.c	2005-06-07 22:50:02.000000000 +0200
+++ vlc-snapshot-20050607-amd64/src/misc/cpu.c	2005-06-08 13:05:49.821853128 +0200
@@ -80,15 +80,23 @@
     return i_capabilities;
 
 #elif defined( __i386__ ) || defined( __x86_64__ )
-    volatile unsigned int  i_eax, i_ebx, i_ecx, i_edx;
+    volatile unsigned long i_eax, i_ebx, i_ecx, i_edx;
     volatile vlc_bool_t    b_amd;
 
+#  ifdef __x86_64__
+#    define REG_a "rax"
+#    define REG_b "rbx"
+#  else
+#    define REG_a "eax"
+#    define REG_b "ebx"
+#  endif
+
     /* Needed for x86 CPU capabilities detection */
 #   define cpuid( reg )                    \
-        asm volatile ( "push %%ebx\n\t"   \
+        asm volatile ( "push %%"REG_b"\n\t"   \
                        "cpuid\n\t"         \
-                       "movl %%ebx,%1\n\t" \
-                       "pop %%ebx\n\t"    \
+                       "mov %%"REG_b",%1\n\t" \
+                       "pop %%"REG_b"\n\t"    \
                      : "=a" ( i_eax ),     \
                        "=r" ( i_ebx ),     \
                        "=c" ( i_ecx ),     \
@@ -103,17 +111,17 @@
 
     i_capabilities |= CPU_CAPABILITY_FPU;
 
-    asm volatile ( "push %%ebx\n\t"
+    asm volatile ( "push %%"REG_b"\n\t"
                    "pushf\n\t"
-                   "pop %%eax\n\t"
+                   "pop %%"REG_a"\n\t"
                    "movl %%eax, %%ebx\n\t"
                    "xorl $0x200000, %%eax\n\t"
-                   "push %%eax\n\t"
+                   "push %%"REG_a"\n\t"
                    "popf\n\t"
                    "pushf\n\t"
-                   "pop %%eax\n\t"
-                   "movl %%ebx,%1\n\t"
-                   "pop %%ebx\n\t"
+                   "pop %%"REG_a"\n\t"
+                   "mov %%"REG_b",%1\n\t"
+                   "pop %%"REG_b"\n\t"
                  : "=a" ( i_eax ),
                    "=r" ( i_ebx )
                  :
