Index: vlc-0.8.2/configure.ac
===================================================================
--- vlc-0.8.2.orig/configure.ac
+++ vlc-0.8.2/configure.ac
@@ -1493,6 +1493,9 @@ if test "${enable_smb}" != "no"; then
     [ if test -n "${enable_smb}"; then
         AC_MSG_ERROR([cannot find libsmbclient headers])
      fi ])
+  AC_CHECK_MEMBER([struct _SMBCCTX.close_fn],
+    AC_DEFINE([HAVE__SMBCCTX_CLOSE_FN], [1], [Define if samba has _SMBCCTX.close_fn]),,
+    [#include <libsmbclient.h>])
 fi
 
 dnl
Index: vlc-0.8.2/modules/access/smb.c
===================================================================
--- vlc-0.8.2.orig/modules/access/smb.c
+++ vlc-0.8.2/modules/access/smb.c
@@ -301,7 +301,11 @@ static void Close( vlc_object_t *p_this 
     access_sys_t *p_sys = p_access->p_sys;
 
 #ifdef USE_CTX
+#  ifndef HAVE__SMBCCTX_CLOSE_FN
     p_sys->p_smb->close( p_sys->p_smb, p_sys->p_file );
+#  else
+    p_sys->p_smb->close_fn( p_sys->p_smb, p_sys->p_file );
+#  endif
     smbc_free_context( p_sys->p_smb, 1 );
 #else
     smbc_close( p_sys->i_smb );
