--- vlc-1.1.13/modules/video_filter/Modules.am~	2011-10-07 22:31:47.000000000 +0200
+++ vlc-1.1.13/modules/video_filter/Modules.am	2011-12-22 11:07:00.437871107 +0100
@@ -23,8 +23,11 @@
 libosdmenu_plugin_la_CFLAGS = $(AM_CFLAGS) -DPKGDATADIR=\"$(vlcdatadir)\"
 libosdmenu_plugin_la_LIBADD = $(AM_LIBADD)
 libosdmenu_plugin_la_DEPENDENCIES =
+
+if BUILD_OSDMENU
 EXTRA_LTLIBRARIES += libosdmenu_plugin.la
 libvlc_LTLIBRARIES += $(LTLIBosdmenu)
+endif
 
 SOURCES_remoteosd = remoteosd.c remoteosd_rfbproto.h
 SOURCES_magnify = magnify.c
