--- vlc-0.8.1/modules/demux/livedotcom.cpp.orig	2004-11-13 11:55:13.000000000 +0100
+++ vlc-0.8.1/modules/demux/livedotcom.cpp	2005-04-09 15:34:11.000000000 +0200
@@ -35,9 +35,9 @@
 #   include <winsock2.h>
 #endif
 
-#include "BasicUsageEnvironment.hh"
-#include "GroupsockHelper.hh"
-#include "liveMedia.hh"
+#include <liveMedia/BasicUsageEnvironment.hh>
+#include <liveMedia/GroupsockHelper.hh>
+#include <liveMedia/liveMedia.hh>
 
 extern "C" {
 #include "../access/mms/asf.h"  /* Who said ugly ? */
