--- vlc-1.1.0/configure.ac~	2010-06-20 22:17:11.000000000 +0200
+++ vlc-1.1.0/configure.ac	2010-07-08 11:28:14.152064066 +0200
@@ -1601,7 +1601,7 @@
 		  plugins (default enabled)]))
 if test "${enable_lua}" != "no"
 then
-  PKG_CHECK_MODULES(LUA, lua5.1,
+  PKG_CHECK_MODULES(LUA, lua51,
     [ have_lua=yes ],
     [
     AC_MSG_WARN([lua5.1 not found, trying lua >= 5.1 instead])
@@ -1631,7 +1631,7 @@
       AC_MSG_ERROR([Could not find lua. Lua is needed for some interfaces (rc, telnet, http) as well as many other custom scripts. Use --disable-lua to ignore this error.])
   fi
   AC_ARG_VAR([LUAC], [LUA byte compiler])
-  AC_PATH_PROGS(LUAC,[${LUAC} luac], [false])
+  AC_PATH_PROGS(LUAC,[${LUAC} luac51], [false])
   AS_IF([test "${ac_cv_path_LUAC}" = "false"], [
     AC_MSG_ERROR([Could not find the LUA byte compiler.])
   ])
